rewrite-java/src/main/java/org/openrewrite/java/internal/DefaultJavaTypeSignatureBuilder.java

```diff
@@ -25,7 +25,7 @@
     public static final DefaultJavaTypeSignatureBuilder TO_STRING = new DefaultJavaTypeSignatureBuilder();
 
     @Nullable
-    private Set<String> typeVariableNameStack;
+    private Set<List<JavaType>> typeVariableNameStack;
```

Comment:
The `List` is a placeholder for peer review and resolves signatures correctly.

Previously, we relied on the type variable name to prevent SOEs on generic recursive types.
The signature was closed with "}" and returned if we saw the same name again.

Example of a generic recurssion: `<U extends JavaTypeGoat<U, ?>> void genericRecursive(JavaTypeGoat<? extends U[], ?> n)`.

I changed the `String` to a `List<JavaType` so that signatures are fully produced without infinite recursion.

Reply:
@traceyyoshima why would it be a `Set<List<JavaType>>` instead of a `Set<JavaType>` (or even better a `List<JavaType>` given that we don't have hash code implementations on `JavaType`?

Reply:
The `Set<List<JavaType>>` was only in place to get input so that I may update the code.
Updated to use a `List<JavaType>`